src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractPackage`2.cs

```diff
@@ -95,17 +96,17 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
                 Workspace.AdviseSolutionEvents(solution);
             }
 
-            LoadComponentsInUIContextOnceSolutionFullyLoaded(cancellationToken);
+            LoadComponentsInUIContextOnceSolutionFullyLoadedAsync(cancellationToken).Forget();
         }
 
-        protected override void LoadComponentsInUIContext(CancellationToken cancellationToken)
+        protected override async Task LoadComponentsInUIContextAsync(CancellationToken cancellationToken)
         {
-            ForegroundObject.AssertIsForeground();
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
❗️ Pass `cancellationToken`

Reply:
if we switch here, why did the caller also switch?

Reply:
In this case, the caller does not need to switch because it only invokes an asynchronous method, and does not have any affinity itself.

In general, characteristics of the caller are never a consideration for the switching characteristics of the callee.